libraries/ESP8266WebServer/src/ESP8266WebServer-impl.h

```diff
@@ -547,97 +517,97 @@ void ESP8266WebServerTemplate<ServerType>::_streamFileCore(const size_t fileSize
 
 template <typename ServerType>
 const String& ESP8266WebServerTemplate<ServerType>::arg(const String& name) const {
-  for (int j = 0; j < _postArgsLen; ++j) {
-    if ( _postArgs[j].key == name )
-      return _postArgs[j].value;
-  }
-  for (int i = 0; i < _currentArgCount; ++i) {
-    if ( _currentArgs[i].key == name )
-      return _currentArgs[i].value;
-  }
-  return emptyString;
+    for (size_t j = 0; j < _postArgsLen; ++j) {
+        if ( _postArgs[j].key == name )
+            return _postArgs[j].value;
+    }
+    for (size_t i = 0; i < _currentArgCount; ++i) {
+        if ( _currentArgs[i].key == name )
+            return _currentArgs[i].value;
+    }
+    return emptyString;
 }
 
 template <typename ServerType>
-const String& ESP8266WebServerTemplate<ServerType>::arg(int i) const {
-  if (i >= 0 && i < _currentArgCount)
-    return _currentArgs[i].value;
-  return emptyString;
+const String& ESP8266WebServerTemplate<ServerType>::arg(size_t i) const {
```

Comment:
Signature change, this could be a breaking change. Either we wait for v3 or add a forwarder method.

Reply:
it is fully source compatible because signed integers can always be numerically promoted to unsigned integers.

Reply:
Nevertheless, it can be a breaking change. Trivial example:
```cpp
struct Base
{
    void method(int a) {std::cout << "a=" << a << std::endl;}    
};

struct Derived : Base
{
//    void method(int a) 
    void method(size_t a) 
    {
        char buffer[128] = {0};      
        sprintf(buffer, "%d", a);
        std::cout << "buffer=" << buffer << std::endl;
        Base::method(a);
    }
};


int main()
{
    Base base;
    Derived derived;
    
    base.method(3);
    derived.method(-3);
}
```

The above will cause a warning, and if building with extra it will fail to compile. Similar will happen if a user has a derived class with a check like `if (a < 0)`, which will also give a warning.
Then, although far less likely, there's a whole range of std::numeric_limits::blah() that could go wrong, as well as template stuff I don't really want to go into detail right now, which could result in straight up compilation errors.

Please beware signature changes. I agree with your intent, and I welcome the proposed change, but merging must be done with great care. We've been bitten by signature changes in the past.
The one place where a signature change is ok is in private methods, because those methods can't be called externally nor from derived classes. And even then there's friend objects to consider, if any.